external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaDataConsumer.scala

```diff
@@ -269,9 +300,12 @@ private[kafka010] case class InternalKafkaConsumer(
           // When there is some error thrown, it's better to use a new consumer to drop all cached
           // states in the old consumer. We don't need to worry about the performance because this
           // is not a common path.
-          resetConsumer()
-          reportDataLoss(failOnDataLoss, s"Cannot fetch offset $toFetchOffset", e)
-          toFetchOffset = getEarliestAvailableOffsetBetween(toFetchOffset, untilOffset)
+          releaseConsumer()
+          fetchedData.reset()
```

Comment:
Don't we need `releaseFetchedData` here?

Reply:
Yes, as FetchedData is designed to be modified per task. Once you get the one for that task, you can just modify it, and also reset if necessary.

Reply:
Had a refreshing look and see the concept, looks OK.